src/exchanges/bittrex/BittrexFeed.ts

```diff
@@ -46,38 +46,60 @@ export class BittrexFeed extends ExchangeFeed {
         return 'Bittrex';
     }
 
-    subscribe(products: string[]): boolean {
+    subscribe(products: string[]): Promise<boolean> {
```

Comment:
If we make the `subscribe` function an "all-or-nothing" affair, then we can guarantee that the array of promises only resolve if they all return true. We can then skip the checks and simplify the `then` part of the logic.